sdk/webpubsub/Microsoft.Azure.WebJobs.Extensions.WebPubSub/src/Utilities.cs

```diff
@@ -218,23 +219,23 @@ public static bool IsValidationRequest(this HttpRequestMessage req, out List<str
             return false;
         }
 
-        private static Dictionary<string, object> GetStatesFromJson(JsonDocument converted, string originStr)
+        private static Dictionary<string, object> GetStatesFromJson(JObject response)
         {
-            if (converted.RootElement.TryGetProperty("states", out var val))
+            var states = new Dictionary<string, object>();
+            if (response.TryGetValue("states", out var val))
             {
-                if (val.ValueKind == JsonValueKind.Object)
+                // val should be a JSON object of <key,value> pairs
+                if (val.Type == JTokenType.Object)
                 {
-                    return JsonSerializer.Deserialize<StatesEntity>(originStr).States;
+                    var strongType = JsonConvert.DeserializeObject<IReadOnlyDictionary<string, BinaryData>>(val.ToString());
```

Comment:
Do you have to `ToString()` it again when the JSON is already parsed?  Does `DeserializeObject` have an overload taking `val`?

Reply:
use `val.ToObject<IReadOnlyDictionary<string, BinaryData>`?